### PR TITLE
Implementation of #44

### PR DESCRIPTION
**Pull request by cyberone**

### CODE REVIEW

**Comment by dmarkov** [comment]
Many thanks for the PR, let me find a reviewer for it

**Comment by dmarkov** [comment]
@longtimeago could you please review this pull request

File: src/test/java/com/jcabi/log/ObjectDecorTest.java [hunk 1]
```diff
@@ -63,28 +63,21 @@ public ObjectDecorTest(final Object object, final String text,
...
-     *  members will also show its internal contents. Ensure that this is
-     *  handled in a "deep" fashion; that is, if its members are also arrays,
-     *  then print the contents of the members of that array member (and so on).
-     *  When done, uncomment the last test case below.
      */
     @Parameters
     public static Collection<Object[]> params() {
         return Arrays.asList(
-            new Object[][] {
-                {null, "NULL", 0, 0, 0},
```
Comment (longtimeago) [inline src/test/java/com/jcabi/log/ObjectDecorTest.java]: why did you remove these 3 lines? These are parameters for another test runs

File: src/main/java/com/jcabi/log/ObjectDecor.java [hunk 2]
```diff
@@ -72,41 +69,18 @@ public void formatTo(final Formatter formatter, final int flags,
...
+                    new ObjectContentsFormatAction(this.object)
+                );
             formatter.format(output);
         }
     }
 
-    /**
-     * {@link PrivilegedAction} for obtaining object contents.
-     */
-    private final class ObjectContentsFormatAction
```
Comment (longtimeago) [inline src/main/java/com/jcabi/log/ObjectDecor.java]: why did you decide to extract this class in separate file? If there is no weighty reason of doing this we need to keep it inside, as it was before

File: src/main/java/com/jcabi/log/ArrayFormatAction.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    ArrayFormatAction(final Object[] arr) {
+        this.array = Arrays.copyOf(arr, arr.length);
+    }
+
+    @Override
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
+    public String run() {
+        final StringBuilder builder = new StringBuilder("[");
+        final Formatter formatter = new Formatter(builder);
+        for (int index = 0; index < this.array.length; index += 1) {
```
Comment (longtimeago) [inline src/main/java/com/jcabi/log/ArrayFormatAction.java]: why don't use same approach as in ObjectContentsFormatAction?
I mean, foreach and 
```builder.replace(...);``` in the end?
It should look better than indexes

**Comment by cyberone** [comment]
@longtimeago please see again

File: src/main/java/com/jcabi/log/ObjectDecor.java [hunk 4]
```diff
@@ -72,41 +73,100 @@ public void formatTo(final Formatter formatter, final int flags,
...
+
+    /**
      * {@link PrivilegedAction} for obtaining object contents.
+     * @author Marina Kosenko (marina.kosenko@gmail.com)
+     * @author Yegor Bugayenko (yegor@tpc2.com)
+     * @version $Id$
      */
-    private final class ObjectContentsFormatAction
+    @Immutable
+    static final class ObjectContentsFormatAction
```
Comment (longtimeago) [inline src/main/java/com/jcabi/log/ObjectDecor.java]: why not private?

File: src/main/java/com/jcabi/log/ObjectDecor.java [hunk 5]
```diff
@@ -72,41 +73,100 @@ public void formatTo(final Formatter formatter, final int flags,
...
         }
     }
 
     /**
+     * {@link PrivilegedAction} for obtaining array contents.
+     * @author Aleksey Popov (alopen@yandex.ru)
+     * @version $Id$
+     */
+    @Immutable
+    static final class ArrayFormatAction
```
Comment (longtimeago) [inline src/main/java/com/jcabi/log/ObjectDecor.java]: why not private?

**Comment by cyberone** [comment]
@longtimeago oops, sorry, fixed.

**Comment by cyberone** [comment]
@longtimeago please see again

File: src/test/java/com/jcabi/log/ObjectDecorTest.java [hunk 6]
```diff
@@ -77,13 +71,13 @@ public ObjectDecorTest(final Object object, final String text,
...
-            //  @checkstyle MethodBodyComments (6 lines)
-            //  @checkstyle LineLength (3 lines)
-            //  {
-            //      new Object[]{new Foo(0, "zero"), new Foo(2, "two")},
-            //      "[{num: \"0\", name: \"zero\"}, \"{num: \"2\", name: \"two\"}]",
-            //      0, 0, 0
-            //  },
+                //  @checkstyle MethodBodyComments (6 lines)
+                //  @checkstyle LineLength (3 lines)
+                {
```
Comment (longtimeago) [inline src/test/java/com/jcabi/log/ObjectDecorTest.java]: one more test example is missing - array, which contains another array. This case described in your task description

**Comment by cyberone** [comment]
@longtimeago done, please re-check though there was no need in this test because it was enough to be sure we're not using deepToString() at ArrayFormatAction.

**Comment by longtimeago** [comment]
@cyberone now it's good to merge, thank you!

**Comment by longtimeago** [comment]
@rultor please, merge

**Comment by longtimeago** [comment]
@cyberone you need to merge back from master

**Comment by cyberone** [comment]
@longtimeago re-run rultor please

**Comment by longtimeago** [comment]
@rultor please, merge again

**Comment by dmarkov** [comment]
@longtimeago done, I added **17 mins** in payment `50616606`

+17 added to your rating, at the moment it is: [+1401](http://www.netbout.com/b/35024?open=rating)

**Comment by dmarkov** [comment]
@rultor deploy
